File: topological.py
from collections import defaultdict

class Graph:

    def __init__(self,n):

        self.graph = defaultdict(list)

        self.N = n

    def addEdge(self,m,n):

        self.graph[m].append(n)

    def sortUtil(self,n,visited,stack):

        visited[n] = True

        for element in self.graph[n]:

            if visited[element] == False:

                self.sortUtil(element,visited,stack)

        stack.insert(0,n)

    def topological(self):

        visited = [False]*self.N

        stack =[]

        for element in range(self.N):

            if visited[element] == False:

                self.sortUtil(element,visited,stack)

        print(stack)

graph = Graph(5)
graph.addEdge(0,2)
graph.addEdge(1,4)
graph.addEdge(2,3)
graph.addEdge(3,2)
graph.addEdge(4,1)

print("The Topological Sort Of The Graph Is:  ")

graph.topological()

File: topological.cpp
#include <iostream>
#include <list>
#include <stack>
using namespace std;

class Graph {
	int M; 

	list<int>* adj;

	void topologicalSort(int m, bool visited[], stack<int>& Stack);

public:
	Graph(int M); 

	void addEdge(int m, int n);

	void topological();
};

Graph::Graph(int M)
{
	this->M = M;
	adj = new list<int>[M];
}

void Graph::addEdge(int m, int n)
{
	adj[m].push_back(n); 
}

void Graph::topologicalSort(int m, bool visited[],stack<int>& Stack)
{
	visited[m] = true;

	list<int>::iterator i;
	for (i = adj[m].begin(); i != adj[m].end(); ++i)
		if (!visited[*i])
			topologicalSort(*i, visited, Stack);

	Stack.push(m);
}

void Graph::topological()
{
	stack<int> Stack;

	bool* visited = new bool[M];
	for (int i = 0; i < M; i++)
		visited[i] = false;

	for (int i = 0; i < M; i++)
		if (visited[i] == false)
			topologicalSort(i, visited, Stack);

	while (Stack.empty() == false) {
		cout << Stack.top() << " ";
		Stack.pop();
	}
}

int main()
{
	Graph g(5);
	g.addEdge(3, 1);
	g.addEdge(4, 3);
	g.addEdge(0, 2);
	g.addEdge(2, 3);
	g.addEdge(1, 2);

	cout << "Following is a Topological Sort of the given graph: \n ";
	g.topological();

	return 0;
}